<launch>

        <arg name="point_topic" default="/points_raw" />
        <arg name="no_ground_point_topic" default="/points_no_ground" />
        <arg name="ground_point_topic" default="/points_ground" />

        <arg name="sensor_model" default="64" />
        <arg name="sensor_height" default="1.73" />
        <arg name="max_slope" default="5.0" />
        <arg name="vertical_thres" default="0.08" />

        <arg name="remove_floor" default="true" />

        <arg name="plane_dis_thres" default="0.35" />
        <arg name="n_section" default="4" />
        <arg name="plane_height_thres" default="5.0" />
        <arg name="plane_ang_thres" default="16.0" />
        <arg name="save_flag" default="true"/>
	<arg name="save_csv" default="true"/>
	<arg name="pcd_savepath" default="/media/gky/Elements5/数据集/results/ground_seg/cascadedSeg/pcd/10" />
	<arg name="output_filename" default="/media/gky/Elements5/数据集/results/ground_seg/cascadedSeg/csv/10/cascadedSeg.csv" />
	<param name="data_path" value="/media/gky/Elements5/data_all/kitti数据集/semantickitti/dataset/sequences/10" />

	<node pkg="cascaded_ground_seg" type="offline_gs" name="cascaded_ground_seg" output="screen">
                <param name="point_topic" value="$(arg point_topic)" />
                <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)" />
                <param name="ground_point_topic" value="$(arg ground_point_topic)" />

                <param name="sensor_model" value="$(arg sensor_model)" />
                <param name="sensor_height" value="$(arg sensor_height)" />
                <param name="max_slope" value="$(arg max_slope)" />
                <param name="vertical_thres" value="$(arg vertical_thres)" />

                <param name="remove_floor" value="$(arg remove_floor)" />

                <param name="plane_dis_thres" value="$(arg plane_dis_thres)" />
                <param name="n_section" value="$(arg n_section)" />
                <param name="plane_height_thres" value="$(arg plane_height_thres)" />
                <param name="plane_ang_thres" value="$(arg plane_ang_thres)" />
		<param name="save_flag" value="$(arg save_flag)"/>
		<param name="pcd_savepath" value="$(arg pcd_savepath)"/>
		<param name="save_csv" value="$(arg save_csv)"/>
		<param name="output_filename" value="$(arg output_filename)"/>
        </node>

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find cascaded_ground_seg)/rviz/patchwork_viz.rviz"/>

</launch>
